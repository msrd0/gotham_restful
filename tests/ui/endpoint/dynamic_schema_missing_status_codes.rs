#[macro_use]
extern crate gotham_restful;

use gotham::hyper::StatusCode;
use gotham_restful::private::OpenapiSchema;

#[derive(Resource)]
#[resource(read_all)]
struct FooResource;

fn schema(_: StatusCode) -> OpenapiSchema {
	unimplemented!()
}

#[read_all(schema = "schema")]
async fn read_all() {}

fn main() {}
