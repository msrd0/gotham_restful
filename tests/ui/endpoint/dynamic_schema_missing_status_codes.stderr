error: `schema` and `status_codes` may only be used together
  --> tests/ui/endpoint/dynamic_schema_missing_status_codes.rs:15:21
   |
15 | #[read_all(schema = "schema")]
   |                     ^^^^^^^^

error[E0412]: cannot find type `read_all___gotham_restful_endpoint` in this scope
 --> tests/ui/endpoint/dynamic_schema_missing_status_codes.rs:8:12
  |
8 | #[resource(read_all)]
  |            ^^^^^^^^ not found in this scope
  |
help: you might be missing a type parameter
  |
7 | #[derive(Resource<read_all___gotham_restful_endpoint>)]
  |                  ++++++++++++++++++++++++++++++++++++
